deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/ClusterReplacer.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2016 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pinterest.teletraan.worker;
+
+
+import com.pinterest.arcee.autoscaling.AutoScalingManager;
+import com.pinterest.clusterservice.bean.ClusterBean;
+import com.pinterest.clusterservice.bean.ClusterState;
+import com.pinterest.clusterservice.dao.ClusterDAO;
+import com.pinterest.clusterservice.handler.ClusterHandler;
+import com.pinterest.deployservice.ServiceContext;
+import com.pinterest.deployservice.bean.EnvironBean;
+import com.pinterest.deployservice.bean.HostBean;
+import com.pinterest.deployservice.dao.AgentDAO;
+import com.pinterest.deployservice.dao.EnvironDAO;
+import com.pinterest.deployservice.dao.HostDAO;
+import com.pinterest.deployservice.dao.UtilDAO;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+public class ClusterReplacer implements Runnable {
+    private static final Logger LOG = LoggerFactory.getLogger(ClusterReplacer.class);
+    private final AgentDAO agentDAO;
+    private final ClusterDAO clusterDAO;
+    private final EnvironDAO environDAO;
+    private final HostDAO hostDAO;
+    private final UtilDAO utilDAO;
+    private final ClusterHandler clusterHandler;
+    private final AutoScalingManager autoScalingManager;
+
+    public ClusterReplacer(ServiceContext serviceContext) {
+        agentDAO = serviceContext.getAgentDAO();
+        clusterDAO = serviceContext.getClusterDAO();
+        environDAO = serviceContext.getEnvironDAO();
+        hostDAO = serviceContext.getHostDAO();
+        utilDAO = serviceContext.getUtilDAO();
+        clusterHandler = new ClusterHandler(serviceContext);
+        autoScalingManager = serviceContext.getAutoScalingManager();
+    }
+
+    private void replaceCluster(ClusterBean clusterBean) throws Exception {
+        String clusterName = clusterBean.getCluster_name();
+        List<EnvironBean> envs = environDAO.getEnvsByGroups(Collections.singletonList(clusterName));
+        EnvironBean envBean = null;
+        for (EnvironBean env : envs) {
+            // Find the env/stage
+            if (clusterName.equals(clusterHandler.getClusterName(env.getEnv_name(), env.getStage_name()))) {
+                envBean = env;
+                break;
+            }
+        }
```

Comment:
If we think there is a one-to-one mapping, we could just use cluste rname and derive the env/stage name.